packages/bpk-component-drawer/src/BpkDrawerContent.js

```diff
@@ -0,0 +1,121 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2017 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import Transition from 'react-transition-group/Transition';
+import { animations } from 'bpk-tokens/tokens/base.es6';
+import { cssModules } from 'bpk-react-utils';
+import { BpkButtonLink } from 'bpk-component-link';
+import BpkCloseButton from 'bpk-component-close-button';
+
+import STYLES from './bpk-drawer-content.scss';
+
+const getClassName = cssModules(STYLES);
+
+const BpkDrawerContent = (props) => {
+  const contentClassNames = [getClassName('bpk-drawer')];
+  const headerClassNames = [getClassName('bpk-drawer__heading')];
+
+  if (props.className) {
+    contentClassNames.push(props.className);
+  }
+
+  if (props.hideTitle) {
+    headerClassNames.push(getClassName('bpk-drawer__heading--visually-hidden'));
+  }
+
+  const headingId = `bpk-drawer-heading-${props.id}`;
+
+  return (
+    <Transition
+      timeout={{
+        enter: 0,
+        exit: parseInt(animations.durationSm, 10),
+      }}
+      appear
+      enter={false}
+      exit
+      in={props.isDrawerShown}
+      onExited={props.onCloseAnimationComplete}
+    >
+      {status => (
+        <section
+          id={props.id}
+          tabIndex="-1"
+          role="dialog"
+          key="dialog"
+          aria-labelledby={headingId}
+          className={[contentClassNames.join(' '), getClassName(`bpk-drawer--${status}`)].join(' ')}
+          ref={props.getDialogRef}
+          {...props.closeEvents}
+        >
+          <header className={getClassName('bpk-drawer__header')}>
+            <h2 id={headingId} className={headerClassNames.join(' ')}>
+              {props.title}
+            </h2>
+            &nbsp;
+            {props.closeText
+              ? <BpkButtonLink onClick={props.onClose}>{props.closeText}</BpkButtonLink>
+              : <BpkCloseButton
+                className={getClassName('bpk-drawer__close-button')}
+                label={props.closeLabel}
+                onClick={props.onClose}
+              />
+            }
+          </header>
+          <div className={getClassName('bpk-drawer__content')}>
+            {props.children}
+          </div>
+        </section>
+        )}
+    </Transition>
+  );
+};
+
+BpkDrawerContent.propTypes = {
+  getDialogRef: PropTypes.func.isRequired,
```

Comment:
We typically don't prefix ref functions with `get`. So this would just be `dialogRef`. Why is it required?

Reply:
I can rename it but it's something the scrim passes in, so it'll need a bit more work there.

dialogRef is used to give focus to the thing on top of the scrim so that screen readers know what's going on.

Reply:
If you're really wanting the name changed I can do it in a follow-up PR that touches the scrim, the modal, and this.

Reply:
Hmm yeah let's defer it. Right I guess it makes sense since this is internal only